src/routes/+page.svelte

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+    import { onMount } from 'svelte';
+    import Gallery from '../components/Gallery.svelte';
+    import MovieDetail from '../components/MovieDetail.svelte';
+    import NavBar from '../components/NavBar.svelte';
+    import '/src/global.css';
+    import type { Movie } from '../interfaces/Movie';
+    import { page } from '$app/stores';
+    import { goto } from '$app/navigation';
+
+    let movies: Movie[] = [];
+    let currentIndex = -1;
+    let showDetail = false;
+    let currentMovie: Movie | undefined;
+    let franchiseId: number;
+    let error: string | null = null;
+
+    $: {
+        const { movies: loadedMovies } = $page.data;
+        if (loadedMovies) {
+            movies = loadedMovies;
+            error = null;
+        } else {
+            error = "No movies loaded";
+            console.log(error);
+        }
+    }
+
+    function nextMovie() {
+        setInterval(() => {
+            if (!showDetail) {
+                currentIndex = (currentIndex + 1) % movies.length;
+                currentMovie = movies[currentIndex];
+            }
+            showDetail = !showDetail;
+        }, 20000);
+    }
+
+    function backToGallery() {
+        showDetail = false;
+        currentMovie = undefined;
+    }
+
+    function handleFranchiseSelected(event: CustomEvent<{ id: number }>) {
+        franchiseId = event.detail.id;
+        goto(`/?franchiseId=${franchiseId}`);
+    }
+
+    onMount(() => {
+        setTimeout(() => {
+            nextMovie();
+        }, 7000);
+    });
+</script>
+
+<main>
+    <NavBar on:franchiseSelected={handleFranchiseSelected} />
+    <div class="content">
+        {#if error}
+            <p class="error">{error}</p>
+        {/if}
+        {#if showDetail && currentMovie}
+            <MovieDetail movie={currentMovie} on:backToGallery={backToGallery} />
+        {/if}
+        <Gallery {movies} on:showDetail={(e) => { currentMovie = e.detail.movie; showDetail = true; }} />
```

Comment:
This would be a bit more elegant in a `handleShowDetail` function I think.

Reply:
This is a good idea indeed and pretty simple to implement this looks better :

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-1e1270da740e81dd13f8a65057582b0fce8a5fb3817ef913f3e90d7f82c4e9df